### PR TITLE
Add document about optimization with `ctr-remote`

### PR DESCRIPTION
**Pull request by ktock**
This includes updates for the demo environment according to the contents of the doc.

### CODE REVIEW

File: docs/ctr-remote.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+## Mounting files from the host
+
+There are several cases where sharing files from host to the container during optimization is useful.
+This includes when we want to optimize an image against building a binary using compilers.
+
+For these use-cases, you can mount files on the hosts to the container using `--mount` option.
+The following example optimizes the image against the workload where Go compiler compiles "hello world" program.
+
+First, create the following Go source file at `/etc/hello.go`.
```
Comment (AkihiroSuda) [inline docs/ctr-remote.md:100]: Having a Go file under `/etc` looks weird

Comment (AkihiroSuda) [reply docs/ctr-remote.md:100]: I guess you meant /tmp/hello.go here

File: docs/ctr-remote.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
+# Optimize Images with `ctr-remote image optimize`
+
+This doc describes example usages of `ctr-remote image optimize` command for converting images into eStargz.
+
+`ctr-remote images optimize` command (call `ctr-remote` in this doc) enables users to convert an image into eStargz.
+The converted image can be lazily pulled by Stargz Snapshotter which can speed up the container startup.
+Because this image is backward compatible to OCI/Docker image, this can be also run by other runtimes that don't support lazy pull (e.g. Docker).
+
+Though lazy pull speeds up the container's startup, it's possible, especially with slow NW, that the runtime performance becomes lower because reading files can induce remotely downloading file contents.
```
Comment (AkihiroSuda) [inline docs/ctr-remote.md:9]: NW -> network

File: docs/ctr-remote.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+|`--cni-plugin-config-dir`|Path to the directory where CNI plugins configurations are stored|
+|`--cni-plugin-dir`|Path to the directory where CNI plugins binaries are installed|
+
+By default, `/etc/hosts` and `/etc/resolv.conf` are configured as the following.
+
+/etc/hosts
+
+```
+127.0.0.1	localhost
+::1	localhost ip6-localhost ip6-loopback
```
Comment (AkihiroSuda) [inline docs/ctr-remote.md:160]: indent

File: docs/ctr-remote.md (hunk 1 above)
Comment (ktock) [reply docs/ctr-remote.md:100]: Exactly.

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**